workflow/controller/controller.go

```diff
@@ -522,10 +530,9 @@ func (wfc *WorkflowController) processNextPodCleanupItem(ctx context.Context) bo
 	logCtx := log.WithFields(log.Fields{"key": key, "action": action})
 	logCtx.Info("cleaning up pod")
 	err := func() error {
-		pods := wfc.kubeclientset.CoreV1().Pods(namespace)
 		switch action {
 		case terminateContainers:
-			pod, err := wfc.getPod(namespace, podName)
+			pod, err := wfc.getPodFromCache(namespace, podName)
```

Comment:
Why is it safe to use the cache here? 
What if pod.Status.Phase is out of sync with what is really going on? What would happen in that case?

Reply:
Hmm I guess the only thing that will happen is for the deletePod action to happen, which I think should be okay